docs/Displaying a User Interface/Main Screen Templates/index.md

```diff
@@ -134,6 +134,294 @@ if (success) {
+
+For the example below, the ```HomeScreen``` class will inherit the ```CustomSDLScreen``` protocol and will have a property of ```HomeDataViewModel```. The screen manager will change its text fields based on the view model's data. In addition, the home screen will also create a navigation button to open the ```SDLButtonScreen``` when pressed.
+
+@![iOS]
+|~
+```objc
+// HomeSDLScreen.h
+#import <Foundation/Foundation.h>
+#import "CustomSDLScreen.h"
+
+@class HomeDataViewModel;
+@class SDLManager;
+@class SDLScreenManager;
+@class SDLSoftButtonObject;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface HomeSDLScreen : NSObject<CustomSDLScreen>
+
+@end
+
+NS_ASSUME_NONNULL_END
+
+/************************************************************************/
+// HomeSDLScreen.m
+#import "HomeSDLScreen.h"
+
+#import "ButtonSDLScreen.h"
+#import "CustomSDLScreen.h"
+#import "HomeDataViewModel.h"
+#import "SmartDeviceLink.h"
+
+@interface HomeSDLScreen()
+
+@property (strong, nonatomic) SDLManager *sdlManager;
+// A button to navigate to the ButtonSDLScreen
+@property (strong, nonatomic) SDLSoftButtonObject *navigationButton;
+// An example of your data model that will feed data to the SDL screen's UI
+@property (strong, nonatomic) HomeDataViewModel *homeDataViewModel;
+
+@end
+
+@implementation HomeSDLScreen
+
+- (instancetype)initWithManager:(SDLManager *)sdlManager {
+    self = [super init];
+    if (!self) { return nil; }
+
+    _sdlManager = sdlManager;
+    // Set navigation button to an SDLSoftButtonObject that will display the Button SDL Screen
+    _navigationButton = [[SDLSoftButtonObject alloc] initWithName:@"ButtonSDLScreen" text:@"Button Screen" artwork:nil handler:^(SDLOnButtonPress * _Nullable buttonPress, SDLOnButtonEvent * _Nullable buttonEvent) {
+        if (buttonPress == nil) { return; }
+
+        ButtonSDLScreen *buttonScreen = [[ButtonSDLScreen alloc] initWithManager:sdlManager];
+        buttonScreen.showScreen();
+    }];
+    _homeDataViewModel = HomeDataViewModel();
+
+    return self;
+}
+
+- (void)showScreen {
+    // Batch updates
+    [self.sdlManager.screenManager beginUpdates];
+    // Change template to Graphics With Text
+    [self.sdlManager.screenManager changeLayout:[[SDLTemplateConfiguration alloc] initWithTemplate:SDLPredefinedLayoutGraphicWithText] withCompletionHandler:nil];
+    // Assign text fields to view model data
+    self.sdlManager.screenManager.textField1 = self.homeDataViewModel.text1;
+    self.sdlManager.screenManager.textField2 = self.homeDataViewModel.text2;
+    self.sdlManager.screenManager.textField3 = self.homeDataViewModel.text3;
+    self.sdlManager.screenManager.textField4 = self.homeDataViewModel.text4;
+    // Assign to navigation button property
+    self.sdlManager.screenManager.softButtonObjects = @[self.navigationButton];
```

Comment:
I don't think you need to create and hold `self.navigationButton` in the `init`. You can create it in this method (or create a function like `- (SDLSoftButtonObject *)createNavigationButton { }`